### ISSUE TITLE
Backspace not working as expected in a TextField with Linux ncurses

### ISSUE THREAD

**Issue by XorZy** [migueldeicaza/gui.cs#1525]
### Problem

On Manjaro Linux with the ncurses driver the backspace key behaves like the delete key in a TextField and deletes the next character instead of the previous.
The issue is not present  with the System Console Driver, but the latter has rendering issues which makes it impossible to use as a workaround.
As far as I know KeyCode 127 (confusingly named Key.Delete) should be interpreted as a backspace and not like the actual Delete Key.
 
### Suggested changes

In `TextField.cs`

```cs
	public override bool ProcessKey (KeyEvent kb)
		{
			// remember current cursor position
			// because the new calculated cursor position is needed to be set BEFORE the change event is triggest
			// Needed for the Elmish Wrapper issue https://github.com/DieselMeister/Terminal.Gui.Elmish/issues/2
			var oldCursorPos = point;

			switch (ShortcutHelper.GetModifiersKey (kb)) {
			-case Key.Delete:
			case Key.DeleteChar:
			case Key.D | Key.CtrlMask:
				if (ReadOnly)
					return true;

				if (length == 0) {
					if (text.Count == 0 || text.Count == point)
						return true;

					SetText (text.GetRange (0, point).Concat (text.GetRange (point + 1, text.Count - (point + 1))));
					Adjust ();
				} else {
					DeleteSelectedText ();
				}
				break;
                       +case Key.Delete:
			case Key.Backspace:
				if (ReadOnly)
					return true;
```

**Comment by BDisp** [comment]
There is a need to take care this, because is specific to some Linux versions and not to all versions.

**Comment by XorZy** [comment]
Well, given the name, that was my initial thought as well.
But after reading it is my understanding that, although it is not the preferred way to represent a BS, it is interpreted as such on a Linux machine.
https://en.wikipedia.org/wiki/Delete_character 
`Unix-like operating systems can still use it as the erase character, i.e., to delete the previous character in the line mode.`

The article also mentions that it was never used on Windows so it should not create any compatibility issue.
I should note that on my end pressing the delete key always returns Key.DeleteChar.
However Backspace can return either Key.Delete when using ncurses or Key.Backspace when using SystemConsole.
However as with all things with Linux there is always room for variation so the most cautious approach would be test multiple terminal emulators and see what they return. If anyone with a Linux machine could report their results I think it would give us a better idea.
Ideally, the driver should be able to detect that automatically, but I'm not even sure that is possible.

**Comment by BDisp** [comment]
The `Delete` key is mapped to the `Key.DeleteChar` on the three drivers. The assigned value of the `Key.DeleteChar` is only a hack to deal with the bitwise flags of the `Key` enum. I did changes there a long time ago and I know that is not the ideal, but for to not breaking anything, I maintained some keys that may not should stay there. It can also have some different behaviors by pressing on the alphanumeric keypad or on the numerical keyboard.
